packages/react-server/core/renderMiddleware.js

```diff
@@ -823,40 +823,48 @@ function writeElements(res, elements) {
 
 	// Pick up where we left off.
 	var start = RLS().nextElement||(RLS().nextElement=0);
+	// We keep a separate count for getAboveTheFoldCount in order to omit RootContainers
+	var elementCount = RLS().nextElementForATF||(RLS().nextElementForATF=0);
 
 	for (var i = start; i < elements.length; RLS().nextElement = ++i){
 
 		// If we haven't rendered the next element yet, we're done.
 		if (elements[i] === ELEMENT_PENDING) break;
 
 		// Got one!
-		writeElement(res, elements[i], i);
+		var shouldCountForATF = writeElement(res, elements[i], i);
 
 		// Free for GC.
 		elements[i] = ELEMENT_ALREADY_WRITTEN;
 
 		if (PageUtil.PageConfig.get('isFragment')) continue;
 
-		if (i === RLS().atfCount - 1){
+		if (elementCount === RLS().atfCount - 1){
 
 			// Okay, we've sent all of our above-the-fold HTML,
 			// now we can let the client start waking nodes up.
 			bootstrapClient(res)
 			for (var j = 0; j <= i; j++){
 				renderScriptsAsync([{ text: `__reactServerClientController.nodeArrival(${j})` }], res)
```

Comment:
Huh... just realized that this will emit `nodeArrival` events for container control elements... Guess that's okay?